### ISSUE TITLE
Relay Pagination: `max_limit` is not respected (on nested fields)

### ISSUE THREAD

**Issue by rsomani95** [MrThearMan/graphene-django-query-optimizer#76]
### I have read the docs thoroughly before making this bug report.

- [X] Yes

### I have read through other open issues, and my issue is not a duplicate.

- [X] Yes

### What version of the library you are using?

0.2.8

### Which python version are you using?

3.12

### What operating system are you on?

Mac

### Description

I wanted to set up a connection node where all the items are returned. The reason why I didn't use the `DjangoListField` here is because the client expects to work with relay IDs, and it's also easy to swap between using / not using pagination for quick prototyping.

With graphene, setting `max_limit=None` ensures that all the items in that connection are returned.

For this test, I set my relay page size to 2 like this:
```python
GRAPHENE = {
    "RELAY_CONNECTION_MAX_LIMIT": 2,
}
```

I tried setting up the model in two ways (with `None`, and a number greater than the max no. of connections), but both didn't work, and the default page size was still being respected:
```python
class HousingCompanyNode(IsTypeOfProxyPatch, DjangoObjectType):
    real_estates = DjangoConnectionField(RealEstateNode)
    developers = DjangoConnectionField(DeveloperNode, max_limit=9999)

class HousingCompanyNode(IsTypeOfProxyPatch, DjangoObjectType):
    real_estates = DjangoConnectionField(RealEstateNode)
    developers = DjangoConnectionField(DeveloperNode, max_limit=None)
```

---

I think the bug is here:
https://github.com/MrThearMan/graphene-django-query-optimizer/blob/9d58c3be48faa6a4551f06c20c3c712dff4f7068/query_optimizer/fields.py#L215-L223

I'm hitting this code path and it overrides the `size` value of `pagination_args`, not respecting the `max_limit`

**Comment by MrThearMan** [comment]
Correction: `max_limit` is respected for top level connections, just not for nested ones. This was a [conscious choice](https://github.com/MrThearMan/graphene-django-query-optimizer/issues/61#issuecomment-1961831433) to make implementing #61 easier at the time.

However, with the addition of `get_filter_info`, this is now possible, and I was able to fix this. Released in [0.2.10](https://github.com/MrThearMan/graphene-django-query-optimizer/releases/tag/v0.2.10).

**Comment by rsomani95** [comment]
@MrThearMan you're right. I'm in the process of reproducing this with the test models, but when `max_limit` is None for nested fields, we seem to be ignoring `order_by` filters on the nested field because we never hit the `paginate_prefetch_queryset` function. 

I will keep you posted as I try to reproduce.
I'm not 100% sure yet because I had to implement some custom `order_by` logic for my use case, which might be interfering with this.

**Comment by rsomani95** [comment]
I found a separate issue while testing this. `totalCount` is broken on nested fields if `max_limit=None`

To reproduce:

```python
class DeveloperNode(IsTypeOfProxyPatch, DjangoObjectType):
    class Meta:
        model = DeveloperProxy
        interfaces = (relay.Node,)
        connection_class = CustomConnection

class HousingCompanyNode(IsTypeOfProxyPatch, DjangoObjectType):
    real_estates = DjangoConnectionField(RealEstateNode)
    developers = DjangoConnectionField(DeveloperNode, max_limit=None)
```

```graphql
{
  pagedHousingCompanies {
    edges {
      node {
        name
        developers {
          totalCount
          edgeCount
          edges {
            node {
              name
            }
          }
        }
      }
    }
  }
}
```

Where the response I get is as follows:
```json
{
  "data": {
    "pagedHousingCompanies": {
      "edges": [
        {
          "node": {
            "name": "Ballard PLC",
            "developers": {
              "totalCount": 0,
              "edgeCount": 3,
              "edges": [
                {
                  "node": {
                    "name": "Daniel Anderson"
                  }
                },
                {
                  "node": {
                    "name": "Nicholas Hobbs"
                  }
                },
                {
                  "node": {
                    "name": "Norman Richardson"
                  }
                }
              ]
            }
          }
        },
        ...
```

**Comment by rsomani95** [comment]
> I'm in the process of reproducing this with the test models, but when max_limit is None for nested fields, we seem to be ignoring order_by filters on the nested field because we never hit the paginate_prefetch_queryset function.

I was wrong. We do hit the `paginate_prefetch_queryset` but exit early before doing the ordering because of this conditional block:
https://github.com/MrThearMan/graphene-django-query-optimizer/blob/65225ed150cc8af7edd60d6066fddf48bfe24c6d/query_optimizer/optimizer.py#L173-L176

I was running into an issue because I've been relying on the manual `order_by` code you've written (and not the `OrderingFilter`, because this lets me order by computed fields and also ensure that `order_by` is run _after_ all other filters -- but that's worth a separate discussion altogether)

However, is this block still relevant given that we already return early if the field is not a connection field [here](https://github.com/MrThearMan/graphene-django-query-optimizer/blob/65225ed150cc8af7edd60d6066fddf48bfe24c6d/query_optimizer/optimizer.py#L160-L162)?

**Comment by MrThearMan** [comment]
> I found a separate issue while testing this. totalCount is broken on nested fields if max_limit=None

I'll reopen the issue to fix this, I already know what the issues is so shouldn't be too difficult.

**Comment by MrThearMan** [comment]
As for the ordering filter, you might be interested in my [CustomOrderingFilter](https://github.com/MrThearMan/graphene-django-extensions/blob/212c46fb9cad9ea0de63911309a574d872cf5538/graphene_django_extensions/filters.py#L166), since it let's you write custom ordering functions. It makes other changes too, like changing `order_by` statements from strings to enums, and changing filters to e.g. `pk` -> `pkAsc` & `-pk` -> `pkDesc`, so its not a drop in replacement, but maybe worth a look.

**Comment by MrThearMan** [comment]
Fixed both issues with `max_limit=None` in [0.3.0](https://github.com/MrThearMan/graphene-django-query-optimizer/releases/tag/v0.3.0). 

However, the fix for limiting with `first` is not as optimal as it could be, since techically we don't need to count the number of rows in the database in order to use the `first` filter when `max_size=None`. The issue is that the existing functions for calculating pagination limits expect the size to be known, and I didn't want to add yet another logic path for this case (at least not at the moment, since the `paginate_prefetch_queryset` method is getting out of hand as is).

**Comment by rsomani95** [comment]
Awesome!
Yeah, that tradeoff definitely seems reasonable for now
